sdk/keyvault/Azure.Security.KeyVault.Keys/README.md

```diff
@@ -25,96 +25,40 @@ dotnet add package Azure.Security.KeyVault.Keys
-
-    ```PowerShell
-    # Use --hsm-name instead of --name for Managed HSM
-    az keyvault set-policy --name <your-key-vault-name> --spn $Env:AZURE_CLIENT_ID --key-permissions backup delete get list create encrypt decrypt update
-    ```
-
-    > --key-permissions:
-    > Accepted values: backup, create, decrypt, delete, encrypt, get, import, list, purge, recover, restore, sign, unwrapKey, update, verify, wrapKey
-
-    If you have enabled role-based access control (RBAC) for Key Vault instead, you can find roles like "Key Vault Crypto Officer" in our [RBAC guide][rbac_guide].
-    If you are managing your keys using Managed HSM, read about its [access control][access_control] that supports different built-in roles isolated from Azure Resource Manager (ARM).
+```PowerShell
+az keyvault create --hsm-name <your-key-vault-name> --resource-group <your-resource-group-name> --administrators <your-user-object-id> --location <your-azure-location>
+```
 
-* Use the above mentioned Azure Key Vault name to retrieve details of your Vault which also contains your Azure Key Vault URL:
+To get `<your-user-object-id>` you can run the following CLI command:
 
-    ```PowerShell
-    # Use properties.hsmUri instead of properties.vaultUri for Managed HSM
-    az keyvault show --name <your-key-vault-name> --query properties.vaultUri --output tsv
-    ```
+```PowerShell
+az ad user show --id <your-user-principal> --query id
+```
 
-* Create the Azure Key Vault or Managed HSM and grant the above mentioned application authorization to perform administrative operations on the Managed HSM
-(replace `<your-resource-group-name>` and `<your-key-vault-name>` with your own unique names and `<your-service-principal-object-id>` with the value from above):
+### Authenticate the client
 
-If you are creating a standard Key Vault resource, use the following CLI command:
+In order to interact with the Azure Key Vault service, you'll need to create an instance of the [KeyClient][key_client_class] class. You need a **vault url**, which you may see as "DNS Name" in the portal,
+and credentials to instantiate a client object.
 
-```PowerShell
-az keyvault create --resource-group <your-resource-group-name> --name <your-key-vault-name>
-```
+The examples shown below use a [`DefaultAzureCredential`][DefaultAzureCredential], which is appropriate for most scenarios including local development and production environments.
+Additionally, we recommend using a managed identity for authentication in production environments.
+You can find more information on different ways of authenticating and their corresponding credential types in the [Azure Identity][azure_identity] documentation.
 
-If you are creating a Managed HSM resource, use the following CLI command:
+To use the `DefaultAzureCredential` provider shown below,
+or other credential providers provided with the Azure SDK, you must first install the Azure.Identity package:
```

Comment:
```suggestion
To use the `DefaultAzureCredential` shown below, or other credential types provided with the Azure Identity library, you must first install the `Azure.Identity` package:
```

Reply:
Markdown elides new lines in consecutive lines when rendered.